### PR TITLE
Implement AdvertsFragment functionality and UI enhancements

### PR DESCRIPTION
**Pull request by ebesalti**
- Enable users to create new adverts.
- List all adverts in a RecyclerView.
- Enable navigation to AdvertDetailsFragment on advert click.
- Display advert creation time in both list items and AdvertDetailsFragment.
- Introduce a helper class for formatting the creation date and time.
- Utilize Glide for efficient loading and display of advert images.
- Refactor:
  - Rename specific classes, fragments, and variables for clarity and consistency.
  - Update various libraries to their latest versions.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,14 +2,19 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools">
 
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
Comment (VitaliyBeletskiy) [inline app/src/main/AndroidManifest.xml:6]: Why is it required? Firebase works without these permissions.

File: app/src/main/java/se/rebeccazadig/bokholken/Adverts/AdvertDetailsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        binding.toolbar.setNavigationOnClickListener {
+            findNavController().popBackStack()
+        }
+        fetchAdvertDetails(args.advertId)
+    }
+
+    private fun fetchAdvertDetails(advertId: String) {
+        val database = Firebase.database
+        val books = database.getReference("Books").child(advertId)
```
Comment (VitaliyBeletskiy) [inline app/src/main/java/se/rebeccazadig/bokholken/Adverts/AdvertDetailsFragment.kt:53]: All data should come from a repo.

File: app/src/main/java/se/rebeccazadig/bokholken/Adverts/AdvertViewModel.kt [hunk 3]
```diff
@@ -1,8 +1,89 @@
...
+    // This LiveData holds the filtered list
+    private val _filteredAdverts = MediatorLiveData<List<Adverts>>()
+    val filteredAdverts: LiveData<List<Adverts>> get() = _filteredAdverts
+
+
+    private val _advertSaveStatus = MutableLiveData<Boolean>()
+    val advertSaveStatus: LiveData<Boolean> get() = _advertSaveStatus
+
+    init {
+        _filteredAdverts.addSource(searchQuery) { filterAdverts() }
```
Comment (VitaliyBeletskiy) [inline app/src/main/java/se/rebeccazadig/bokholken/Adverts/AdvertViewModel.kt:42]: Can be done inside the property. No need to create init {}.

File: app/src/main/java/se/rebeccazadig/bokholken/Adverts/AdvertsAdapter.kt [hunk 4]
```diff
@@ -1,85 +1,86 @@
...
-        imageRef.getBytes(1000000).addOnSuccessListener {
-            var bitmap = BitmapFactory.decodeByteArray(it, 0, it.size)
+            binding.heartIcon.setOnClickListener {
+                onHeartClick(advert)
+            }
 
-            holder.bokBild.setImageBitmap(bitmap)
-        }.addOnFailureListener {
+            advert.adId?.takeIf { it.isNotEmpty() }?.let { adId ->
+                val imageRef = Firebase.storage.reference.child("annonser").child(adId)
```
Comment (VitaliyBeletskiy) [inline app/src/main/java/se/rebeccazadig/bokholken/Adverts/AdvertsAdapter.kt:63]: Adapter should not have reference to Firebase.

File: app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt [hunk 5]
```diff
@@ -1,170 +1,267 @@
...
                 }
             }
+        }
 
-            // val action = SkapaAnnonsFragmentDirections.actionSkapaAnnonsFragmentToAnnonsFragment()
-            // findNavController().navigate(R.id.action_skapaAnnonsFragment_to_annonsFragment)
-            findNavController().popBackStack()
+        viewModel.advertSaveStatus.observe(viewLifecycleOwner) { isSuccessful ->
+            if (isSuccessful) {
+                Toast.makeText(requireContext(), "Advert saved successfully!", Toast.LENGTH_SHORT)
```
Comment (VitaliyBeletskiy) [inline app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt:138]: Hardcoded string

File: app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt [hunk 6]
```diff
@@ -1,170 +1,267 @@
...
-            val baos = ByteArrayOutputStream()
-            annonsbild!!.compress(Bitmap.CompressFormat.JPEG, 80, baos)
-            val data = baos.toByteArray()
-
-            imageRef.putBytes(data).addOnFailureListener {
-                Log.i("PIA11DEBUG", "UPLOAD FAIL")
-            }.addOnSuccessListener {
-                Log.i("PIA11DEBUG", "UPLOAD OK")
+    private fun fetchAdvertDetails(adId: String) {
+        val database = Firebase.database
```
Comment (VitaliyBeletskiy) [inline app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt:156]: Same as before, there should not be a reference to FB from UI.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ebesalti) [reply app/src/main/AndroidManifest.xml:6]: I was gonna test something. Forgot to remove them.

File: app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt [hunk 7]
```diff
@@ -1,170 +1,267 @@
...
 
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        Log.d("FragmentLifecycle", "onAttach")
+    }
```
[the remark this answers is not in the document]
Comment (ebesalti) [reply app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt:59]: it is removed in second commit

File: app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt (hunk 5 above)
Comment (ebesalti) [reply app/src/main/java/se/rebeccazadig/bokholken/Adverts/CreateAdvertFragment.kt:138]: Something is totally weird with whole PR. My latest changes are not these ones

**Pull request merged**